File: modulo_2/cpp/6.cpp
#include <iostream>

using namespace std;

int main() {
    int c;
    cin >> c;

    while (c--) {
        int n;
        cin >> n;

        int board[n];
        for (int i = 0; i < n; i++) {
            cin >> board[i];
        }

        int count = 0;
        int move = 0;
        while (true) {
            count += 1;
            int aux = move;
            move += board[move];

            if (move < 0 || move >= n) {
                cout << count << '\n';
                break;
            } else if (board[move] == 0){
                cout << count << '\n';
                break;
            }
            board[aux] = 0;
        }
    }

    return 0;
}

File: modulo_2/py/6.py
c = int(input())

for _ in range(c):
    n = int(input())
    board = tuple(map(int, input().split()))
    aux = 0
    out = 0

    while True:
        if (aux < 0 or aux >= n) or board[aux] == 0:
            print(out)
            break

        x = aux
        aux += board[aux]
        if x == 0:
            board = (0,) + board[1:]
        elif x == n-1:
            board = board[:n-1] + (0,)
        else:
            board = board[:x] + (0,) + board[x+1:]

        out += 1
